runtime/bcutil/cfreader.c

```diff
@@ -2442,8 +2449,13 @@ j9bcutil_readClassFileBytes(J9PortLibrary *portLib,
 		classfile->accessFlags |= CFR_ACC_ABSTRACT;
 	}
 
+	/* Java 9 Spec: If the ACC_MODULE flag is set in the access_flags item, then no
+	 * other flag in the access_flags item should be set. If other flags are set, then this
+	 * file is not a module.
+	 */
 	if ((vmVersionShifted >= BCT_Java9MajorVersionShifted)
-			&& (J9_ARE_ALL_BITS_SET(classfile->accessFlags, CFR_ACC_MODULE))) {
+			&& (classfile->majorVersion >= 53)
+			&& (classfile->accessFlags == CFR_ACC_MODULE)) {
```

Comment:
I don't understand this change. It seems it should be checking if the ACC_MODULE bit is set as in the original code, but not if only the ACC_MODULE bit is set.

Reply:
In the Java 9 spec a file is considered a module only if ACC_MODULE is the only access flag set. I added this to double check that modules are not loaded like classes and to match Oracles behaviour for this scenario.

Reply:
Understood, but it seems you need to check if the module bit is set in the class, not if its the only bit set. Or is there there a check somewhere else that checks if the module bit is set? The code doesn't match the description in the comment.

Reply:
Ah the case where ACC_MODULE is set but its not an invalid module?

I also found the rule `If the ACC_INTERFACE flag is not set, ACC_MODULE may not be set.` in the new spec. I will add a check for this in that should address that case.